src/features/elementfeature.jl

```diff
@@ -101,25 +101,14 @@ end
 
 function (efd::ElementFeatureDescriptor)(a::AbstractAtoms)
     @assert all([el in encodable_elements(efd) for el in a.elements]) "Feature $(efd.name) cannot encode some element(s) in this structure!"
-    efd.encoder_decoder(efd, a, ENCODE)
+    efd.encoder_decoder(efd, a)
```

Comment:
if it will always have them as fields, why not just directly call `encode_f` here? I feel like it would improve code readability...obviously the callable syntax is still useful to have on the user's end I think...

Reply:
Different `encoder_decoder.encode_f` functions may take different arguments for different `encoder_decoder` types. 

For instance, if we had 
```julia
efd.encoder_decoder.encode_f(efd, a, efd.encoder_decoder.nbins, efd.encoder_decoder.logspaced)`
```
here instead, then this wouldn't work for any other type of `Codec` (which doesn't have `nbins` or `logspaced` as fields) that you might want to attach to an `ElementFeatureDescriptor` (like `SimpleCodec`)

This is  why the responsibility of handling what parameters get passed, and how are delegated to the callable syntax of `encoder_decoder` (`Codec` object attached).

Reply:
I also have an idea to provide a generic interface
```julia
(fd::AbstractFeatureDescriptor)(a::AbstractAtoms) = fd.encoder_decoder(fd, a)
```
so that functions like these no longer have to be defined unless there's some explicit check to be done before encoding (`like ElementFeatureDescriptor checks if all the elements are encodable or not`).
But, I'm not really sure if something like that would be very useful, since I think checks like these might need to be done for essentially every `FeatureDescriptor`.